Classes/Controller/HaikuController.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace T3docs\Examples\Controller;
+
+use T3docs\Examples\Domain\Repository\HaikuRepository;
+use TYPO3\CMS\Core\Service\FlexFormService;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Fluid\View\StandaloneView;
+use TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer;
+
+/**
+ * Class FalExampleController
+ */
+class HaikuController
+{
+    /**
+     * The back-reference to the mother cObj object set at call time
+     */
+    public ContentObjectRenderer $cObj;
+    public array $conf = [];
+    public StandaloneView $view;
+
+    public function __construct(
+        protected readonly HaikuRepository $haikuRepository,
+    ) {
+    }
+
+    /**
+     * This setter is called when the plugin is called from UserContentObject (USER)
+     * via ContentObjectRenderer->callUserFunction().
+     */
+    public function setContentObjectRenderer(ContentObjectRenderer $cObj): void
+    {
+        $this->cObj = $cObj;
+    }
+
+    public function main(string $content, array $conf): string
+    {
+        $this->conf = $conf;
+        $this->loadFlexFormSettings();
+        $this->initView();
+
+        $action = $_REQUEST['tx_examples_haiku']['action'] ?? 'list';
+        switch ($action) {
```

Comment:
switch is ugly. Maybe we can use match instead:

```
return match ($action) {
    'show' => $this->showAction(...),
    'findByTitle' => $this->findByTitleAction(...)
    'list' => $this->listAction(),
    default => $this->errorAction(),
}

(PHP >= 8.0)

Reply:
why should a switch be ugly? Also it is much easier readable...